proto/info/edge_node_cluster.proto

```diff
@@ -0,0 +1,237 @@
+  // Name of the node
+  string name = 1;
+
+  // Status of the node
+  KubeNodeStatus status = 2;
+
+  // Role of the node is server or not
+  bool role_server = 3;
+
+  // Creation Time of the node in the cluster
+  google.protobuf.Timestamp creation_timestamp = 4;
+
+  // Version of the API Server running on the node
+  string api_server_sersion = 5;
+
+  // Internal IP address of the node
+  string internal_ip = 6;
+}
+
+enum StorageHealthStatus {
+  STORAGE_HEALTH_STATUS_UNKNOWN = 0;
+  STORAGE_HEALTH_STATUS_HEALTHY = 1;
+  STORAGE_HEALTH_STATUS_DEGRAEDED_2_REPLICA_AVAILABLE_REPLICATING = 2;
+  STORAGE_HEALTH_STATUS_DEGRADED_2_REPLICA_AVAILABLE_NOT_REPLICATING = 3;
+  STORAGE_HEALTH_STATUS_DEGRADED_1_REPLICA_AVAILABLE_REPLICATING = 4;
+  STORAGE_HEALTH_STATUS_DEGRADED_1_REPLICA_AVAILABLE_NOT_REPLICATING = 5;
+  STORAGE_HEALTH_STATUS_FAILED = 6;
+}
+
+enum StorageVolumeState {
+  STORAGE_VOLUME_STATE_UNKNOWN = 0;
+  STORAGE_VOLUME_STATE_CREATING = 1;
+  STORAGE_VOLUME_STATE_ATTACHED = 2;
+  STORAGE_VOLUME_STATE_DETACHED = 3;
+  STORAGE_VOLUME_STATE_ATTACHING = 4;
+  STORAGE_VOLUME_STATE_DETACHING = 5;
+  STORAGE_VOLUME_STATE_DELETING = 6;
+}
+
+enum StorageVolumeRobustness {
+  STORAGE_VOLUME_ROBUSTNESS_UNKNOWN = 0;
+  STORAGE_VOLUME_ROBUSTNESS_HEALTHY = 1;
+  STORAGE_VOLUME_ROBUSTNESS_DEGRADED = 2;
+  STORAGE_VOLUME_ROBUSTNESS_FAULTED = 3;
+}
+
+enum StorageVolumePVCStatus {
+  STORAGE_VOLUME_PVC_STATUS_UNKNOWN = 0;
+  STORAGE_VOLUME_PVC_STATUS_BOUND = 1;
+  STORAGE_VOLUME_PVC_STATUS_PENDING = 2;
+  STORAGE_VOLUME_PVC_STATUS_AVAILABLE = 3;
+  STORAGE_VOLUME_PVC_STATUS_RELEASED = 4;
+  STORAGE_VOLUME_PVC_STATUS_FAILED = 5;
+}
+
+enum StorageVolumeReplicaStatus {
+  STORAGE_VOLUME_REPLICA_STATUS_UNKNOWN = 0;
+  STORAGE_VOLUME_REPLICA_STATUS_REBUILDING = 1;
+  STORAGE_VOLUME_REPLICA_STATUS_ONLINE = 2;
+  STORAGE_VOLUME_REPLICA_STATUS_FAILED = 3;
+}
+
+message KubePodNameSpaceInfo {
+  // Name of the namespace
+  string name = 1;
+
+  // Number of pods in the namespace
+  uint32 pod_count = 2;
+
+  // Number of pods in the namespace that are running
+  uint32 pod_running_count = 3;
+
+  // Number of pods in the namespace that are pending
+  uint32 pod_pending_count = 4;
+
+  // Number of pods in the namespace that are failed
+  uint32 pod_failed_count = 5;
+
+  // Number of pods in the namespace that are succeeded
+  uint32 pod_succeeded_count = 6;
+}
+
+enum KubePodStatus {
+  KUBE_POD_STATUS_UNKNOWN = 0;
+  KUBE_POD_STATUS_PENDING = 1;
+  KUBE_POD_STATUS_RUNNING = 2;
+  KUBE_POD_STATUS_SUCCEEDED = 3;
+  KUBE_POD_STATUS_CONTAINER_CREATING = 4;
+  KUBE_POD_STATUS_CRASHLOOP_BACKOFF = 5;
+  KUBE_POD_STATUS_ERROR = 6;
+  KUBE_POD_STATUS_EVICTED = 7;
+  KUBE_POD_STATUS_FAILED = 8;
+}
+
+message KubeEVEAppPodInfo {
+  // Name of the EVE application
+  string name = 1;
+
+  // Application Status
+  KubePodStatus status = 2;
+
+  // Restart count of the application
+  uint32 restart_count = 3;
+
+  // Restart time of the application, seconds ago
```

Comment:
Is this comment correct that it is a relative time? If so it would make sense to make this explicit because we have some place where an uptime (in seconds) displays as a time in 1970.

Reply:
fixed comment